instance/models/mixins/openedx_database.py

```diff
@@ -351,20 +352,44 @@ def _get_mongo_settings(self):
         """
         Return dictionary of mongodb settings
         """
-        return {
+        extra_settings = {}
+        if self.mongodb_replica_set:
+            mongodb_servers = MongoDBServer.objects.filter(
+                mongodb_replica_set=self.mongodb_replica_set
+            ).values_list('hostname', flat=True)
```

Comment:
`values_list` still returns a QuerySet object, which doesn't serialise.  To turn it into the list of hostnames that `EDXAPP_MONGO_HOSTS` expects, you need to wrap it in `list()`.